lib/puppet/application/apply.rb

```diff
@@ -164,11 +167,15 @@ def app_defaults
   end
 
   def run_command
-    if options[:catalog]
-      apply
-    else
-      main
+    lock do
+      if options[:catalog]
+        apply
+      else
+        main
+      end
     end
+  rescue Puppet::LockError
+    Puppet.notice "Run of #{client_class} already in progress; skipping  (#{lockfile_path} exists)"
```

Comment:
I'd recommend we add a spec test that expects lock to throw an error, and ensure we get a notice and return exit(1).

This should also be updated so that we `exit 1` (equivalent to https://github.com/puppetlabs/puppet/blob/master/lib/puppet/application/agent.rb#L360-L361).